### ISSUE TITLE
test failure `test_by_module_statement_value`

### ISSUE THREAD

**Issue by FRidh** [PyCQA/pylint#3198]
### Steps to reproduce
1. Checkout repo at 2.4.2
2. Have Python env (issue occurs with 3.7 and 3.8, not tested with other versions)
3. `pytest`

### Current behavior
```
=================================== FAILURES ===================================
________________________ test_by_module_statement_value ________________________

init_linter = Checker 'master' (responsible for 'F0001', 'F0002', 'F0010', 'I0001', 'I0010', 'I0011', 'I0013', 'I0020', 'I0021', 'I0022', 'E0001', 'E0011', 'E0012')

    def test_by_module_statement_value(init_linter):
        """Test "statement" for each module analized of computed correctly."""
        linter = init_linter
        linter.check(os.path.join(os.path.dirname(__file__), "data"))

        for module, module_stats in linter.stats["by_module"].items():

            linter2 = init_linter
            if module == "data":
                linter2.check(os.path.join(os.path.dirname(__file__), "data/__init__.py"))
            else:
                linter2.check(os.path.join(os.path.dirname(__file__), module))

            # Check that the by_module "statement" is equal to the global "statement"
            # computed for that module
>           assert module_stats["statement"] == linter2.stats["statement"]
E           assert 7 == 0

tests/unittest_lint.py:811: AssertionError
```

### Expected behavior
Test to pass.

### ~pylint --version output~
pylint is at 2.4.2.

```
============================= test session starts ==============================
platform linux -- Python 3.8.0, pytest-5.2.1, py-1.8.0, pluggy-0.13.0

```

**Comment by stanislavlevin** [comment]
@FRidh, hi. Today I faced the same issue.
The reason is not obvious at first.
In my case, this is a dot with the pathname:
```python
796     def test_by_module_statement_value(init_linter):
797         """Test "statement" for each module analized of computed correctly."""
798         linter = init_linter
799         linter.check(os.path.join(os.path.dirname(__file__), "data"))
800  
801         for module, module_stats in linter.stats["by_module"].items():
802  
803             linter2 = init_linter
804             if module == "data":
805                 linter2.check(os.path.join(os.path.dirname(__file__), "data/__init__.py"))
(Pdb) os.path.dirname(__file__)
'/usr/src/RPM/BUILD/python3-module-pylint-2.4.2/tests'
(Pdb) p module
'data.suppliermodule_test'
```
Thus, the linter gets on its input the dotted name with module notation.
Then
```python
> /usr/src/RPM/BUILD/python3-module-pylint-2.4.2/.tox/py37/lib/python3/site-packages/pylint/utils/utils.py(143)expand_modules()
-> filepath = modutils.file_from_modpath(modname.split("."))
(Pdb) modutils.file_from_modpath(modname.split("."))
(Pdb) modname.split(".")
['/usr/src/RPM/BUILD/python3-module-pylint-2', '4', '2/tests/data', 'suppliermodule_test']
(Pdb) modutils.file_from_modpath(modname.split("."))
```
Thus, `expand_modules` just doesn't expect the dot name and the in-line PyLinter unable to find this module.
This is the same as:
```python
import pylint.lint
pylint.lint.Run(["/test/dot.name/tests.module"])
```
while the following works:
```python
import pylint.lint
pylint.lint.Run(["/test/notdotname/tests.module"])
```

**Comment by stanislavlevin** [comment]
astroid takes args from pylint and works as expected.
```python
from astroid import modutils
modutils.file_info_from_modpath(["/test/dot.name/tests", "module"])
```
So, in my opinion `expand_modules` doesn't handle properly.

**Comment by PCManticore** [comment]
Thanks @stanislavlevin for the investigation, appreciate it!

**Comment by sandrotosi** [comment]
yep this is still happening with 2.4.4:
```
=================================== FAILURES ===================================
________________________ test_by_module_statement_value ________________________

init_linter = Checker 'master' (responsible for 'F0001', 'F0002', 'F0010', 'I0001', 'I0010', 'I0011', 'I0013', 'I0020', 'I0021', 'I0022', 'E0001', 'E0011', 'E0012')

    def test_by_module_statement_value(init_linter):
        """Test "statement" for each module analized of computed correctly."""
        linter = init_linter
        linter.check(os.path.join(os.path.dirname(__file__), "data"))
    
        for module, module_stats in linter.stats["by_module"].items():
    
            linter2 = init_linter
            if module == "data":
                linter2.check(os.path.join(os.path.dirname(__file__), "data/__init__.py"))
            else:
                linter2.check(os.path.join(os.path.dirname(__file__), module))
    
            # Check that the by_module "statement" is equal to the global "statement"
            # computed for that module
>           assert module_stats["statement"] == linter2.stats["statement"]
E           assert 20 == 0

tests/unittest_lint.py:811: AssertionError

```
